src/ui/FacetValueSuggestions/FacetValueSuggestions.ts

```diff
@@ -193,31 +205,38 @@ export class FacetValueSuggestions extends Component {
     return suggestions || [];
   }
 
-  private async getQuerySuggestionsKeywords(omnibox: Omnibox): Promise<string[]> {
+  private async getQuerySuggestionsKeywords(omnibox: Omnibox): Promise<IQuerySuggestionKeyword[]> {
     if (this.options.useQuerySuggestions && omnibox.suggestionAddon) {
-      const suggestions = (await omnibox.suggestionAddon.getSuggestion()) || [];
-      return suggestions.map(s => s.text);
+      return (
+        (await omnibox.suggestionAddon.getSuggestion()).map(({ text, html }) => ({
```

Comment:
Is the `.map` needed here? Seems like it is just copying the object and I don't think it's necessary?

If it is, I would prefer to store the result of the `await` call in a variable for readability.

Also, is `getSuggestion` async because it performs a network request? If so, we could handle the error case by using a try-catch.

 e.g.
```
try {
    const suggestions = await omnibox.suggestionAddon.getSuggestion() || [] // I don't think the OR condition is needed, unless getSuggestion is returning null/undefined? In the case of an error, we would go to the catch
    return suggestions.map(suggestion => ({...suggestion}))
} catch (e) {
    //log error
    return []
}

```

Reply:
About the `.map`: it's not actually just copying the object, it's creating an `IQuerySuggestionKeyword` from an `IOmniboxSuggestion`, they just happen to have the same name for `text` and `html`. I'll add a `<IQuerySuggestionKeyword>` for the sake of clarity, despite it not actually being necessary for TypeScript.

About storing the await call, I'm going to move it into a const.

About catching errors, `getSuggestion` does perform a network request down the line and never catches the error, so I'm adding a `try {} catch {}` around the call to the endpoint and returning an empty array. I would gladly log the error, but it already gets logged.